common/storage/storage.go

```diff
@@ -0,0 +1,67 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/common"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("state")
+
+const (
+	// WaitTimeForSnapshotEpochCheck is the time to wait before checking the storage epoch
+	WaitTimeForSnapshotEpochCheck = time.Millisecond * 100
+
+	// SnapshotWaitTimeout is the timeout for waiting for the storage epoch to change
+	SnapshotWaitTimeout = time.Minute * 3
+)
+
+// StorageEpochChangeWaitArgs are the args needed for calling the WaitForStorageEpochChange function
+type StorageEpochChangeWaitArgs struct {
+	TrieStorageManager            common.StorageManager
+	Epoch                         uint32
+	WaitTimeForSnapshotEpochCheck time.Duration
+	SnapshotWaitTimeout           time.Duration
+}
+
+// WaitForStorageEpochChange waits for the storage epoch to change to the given epoch
+func WaitForStorageEpochChange(args StorageEpochChangeWaitArgs) error {
+	log.Debug("waiting for storage epoch change", "epoch", args.Epoch, "wait timeout", args.SnapshotWaitTimeout)
+
+	if args.SnapshotWaitTimeout < args.WaitTimeForSnapshotEpochCheck {
+		return fmt.Errorf("timeout (%s) must be greater than wait time between snapshot epoch check (%s)", args.SnapshotWaitTimeout, args.WaitTimeForSnapshotEpochCheck)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), args.SnapshotWaitTimeout)
+	defer cancel()
```

Comment:
this is not needed as there is no separate goroutine where you treat the ctx.Done, that one is in the same routine, so it will never have any effect.

Reply:
as far as I know, it is go idiomatic to call `defer cancel()` (or to call the `cancel()` function based on a logic)  whenever you work with contexts even if the code is written in such a way that it is impossible to cause memory leaks. If Beni removes that cancel call the linter will complain.

Reply:
In this case, the `cancel()` function needs to be called as we might exit the function through L49, L54, or L58 so we can free up the context resources. 
The same thing is valid for the timer used in the function which is done on L43.